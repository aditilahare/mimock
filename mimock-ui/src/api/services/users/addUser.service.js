import { post } from 'api/AxiosClient';
import Bcrypt from 'bcryptjs';

export const addNewUser = async ({ name, userName, password, userRole }) => {
	const salt = Bcrypt.genSaltSync(12);
	const hashedPassword = Bcrypt.hashSync(password, salt);

	return await post(
		'/admin/users',
		{
			name,
			userName,
			userRole,
			password: hashedPassword,
		},
		undefined,
		{ showFullPageLoader: true }
	).catch((err) => Promise.reject(err));
};
