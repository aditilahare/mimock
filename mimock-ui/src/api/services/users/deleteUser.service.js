import { remove } from 'api/AxiosClient';

export const deleteUser = async (userName) => {
	const url = '/admin/users/delete-user?userName=' + userName;

	return await remove(url)
		.then((res) => res.data)
		.catch((err) => Promise.reject(err));
};
