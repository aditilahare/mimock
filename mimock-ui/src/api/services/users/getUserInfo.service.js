import { get } from 'api/AxiosClient';

export const getUserInfo = async (userId) => {
	const url = `/admin/users/user-info?userId=${userId}`;

	return await get(url, { showFullPageLoader: true })
		.then((res) => res.data.userInfo)
		.catch((err) => Promise.reject(err));
};
