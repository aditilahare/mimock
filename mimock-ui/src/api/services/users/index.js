import { getAllUsers } from './getUsers.service';
import { deleteUser } from './deleteUser.service';

export { getAllUsers, deleteUser };
