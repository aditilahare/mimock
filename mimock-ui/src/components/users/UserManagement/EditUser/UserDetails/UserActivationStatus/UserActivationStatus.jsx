import React, { useState, useEffect } from 'react';
import ActivationSwitch from '@material-ui/core/Switch';
import { useIosSwitchStyles } from '@mui-treasury/styles/switch/ios';
import { useRecoilState } from 'recoil';
import editUserDetailsAtom from 'atoms/editUserDetailsAtom';
import useNotification from 'hooks/useNotification';
import {
	notificationTypes,
	notificationPositions,
} from 'constants/notificationConstants';
import { updateUserActivationStatus } from 'services/users/updateUserActivationStatus.service';
import {
	UserActivationStatusWrapper,
	UserActivationStatusLabel,
	UserActivationStatusActions,
	StatusLabel,
} from './UserActivationStatus.style';

function UserActivationStatus() {
	const iosStyles = useIosSwitchStyles();
	const [userInfo, setUserInfo] = useRecoilState(editUserDetailsAtom);
	const [activationStatus, setActivationStatus] = useState(
		userInfo.isUserActive
	);

	useEffect(() => {
		setActivationStatus(userInfo.isUserActive);
	}, [userInfo.isUserActive]);

	const updateActivationStatus = (e) => {
		updateUserActivationStatus(userInfo.userName, e.target.checked)
			.then((res) => {
				const { userName, updatedActivationStatus } = res;
				setActivationStatus(updatedActivationStatus);
				useNotification({
					type: notificationTypes.NOTIFICATION_TYPE_SUCCESS,
					title: updatedActivationStatus
						? 'User activated'
						: 'User deactivated',
					message: `User - ${userName} has been ${
						updatedActivationStatus ? 'activated' : 'deactivated'
					}`,
					position: notificationPositions.NOTIFICATION_POSITION_BOTTOM_RIGHT,
					animationIn: 'animate__slideInRight',
					animationOut: 'animate__slideOutRight',
				});
				setUserInfo({
					...userInfo,
					isUserActive: updatedActivationStatus,
				});
			})
			.catch(() => {
				useNotification({
					type: notificationTypes.NOTIFICATION_TYPE_ERROR,
					title: `Failed to update activation status`,
					message: 'Please try again',
					animationIn: 'animate__bounceIn',
					animationOut: 'animate__bounceOut',
				});
			});
	};

	const { root, switchBase, checked, thumb, track } = iosStyles;

	return (
		<UserActivationStatusWrapper data-testid='edit-user-activation-status'>
			<UserActivationStatusLabel>Activation status</UserActivationStatusLabel>
			<UserActivationStatusActions>
				<ActivationSwitch
					data-testid='activation-switch'
					classes={{ root, switchBase, checked, thumb, track }}
					checked={activationStatus}
					onChange={(e) => {
						updateActivationStatus(e);
					}}
				/>
				<StatusLabel
					data-testid='activation-status-label'
					$isActive={activationStatus}
				>
					{activationStatus ? 'Active' : 'Inactive'}
				</StatusLabel>
			</UserActivationStatusActions>
		</UserActivationStatusWrapper>
	);
}

export default UserActivationStatus;
