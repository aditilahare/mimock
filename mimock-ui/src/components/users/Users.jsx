import React from 'react';
import UserManagement from './UserManagement';

export default function Users() {
	return <UserManagement />;
}
