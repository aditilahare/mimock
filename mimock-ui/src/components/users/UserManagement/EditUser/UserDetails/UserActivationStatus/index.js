import UserActivationStatus from './UserActivationStatus';

export default UserActivationStatus;
