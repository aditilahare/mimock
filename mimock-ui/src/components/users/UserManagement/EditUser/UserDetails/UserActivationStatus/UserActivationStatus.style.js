import tw from 'tailwind-styled-components';
import { Info, Label, DynamicValue } from '../UserDetails.style';

export const UserActivationStatusWrapper = tw(Info)``;

export const UserActivationStatusLabel = tw(Label)``;

export const UserActivationStatusActions = tw(DynamicValue)``;

export const StatusLabel = tw.div`
    font-sans
    font-semibold
    text-xl
    mx-4
    ${(props) => (props.$isActive ? 'text-slate-800' : 'text-slate-300')}
`;
