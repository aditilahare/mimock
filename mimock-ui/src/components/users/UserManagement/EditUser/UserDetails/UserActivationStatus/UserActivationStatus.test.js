import React from 'react';
import { act, render, fireEvent } from '@testing-library/react';
import { useRecoilState } from 'recoil';
import { updateUserActivationStatus } from 'services/users';
import UserActivationStatus from './UserActivationStatus';

let mockedRecoilFn = jest.fn();
jest.mock('recoil');
jest.mock('services/users');

const actualNotification = jest.requireActual('hooks/useNotification');
jest.mock('react-notifications-component', () => {
	const Store = {
		addNotification: jest.fn(),
	};
	return { Store };
});

describe('UserActivationStatus', () => {
	beforeEach(() => {
		updateUserActivationStatus.mockResolvedValue({
			userName: 'test123',
			updatedActivationStatus: true,
		});
		useRecoilState.mockImplementation(() => {
			return [
				{
					userName: 'test123',
					name: 'Test User',
					isUserActive: false,
					userRole: 'ADMIN',
					userCreatedAt: '2022-04-30 20:35:35',
					passwordUpdatedAt: '2022-04-30 20:35:35',
				},
				mockedRecoilFn,
			];
		});
	});

	afterEach(() => {
		jest.clearAllMocks();
	});

	it('should render UserActivationStatus when user status is active', async () => {
		useRecoilState.mockImplementation(() => {
			return [
				{
					userName: 'test123',
					name: 'Test User',
					isUserActive: true,
					userRole: 'ADMIN',
					userCreatedAt: '2022-04-30 20:35:35',
					passwordUpdatedAt: '2022-04-30 20:35:35',
				},
				mockedRecoilFn,
			];
		});
		let tree;
		await act(async () => {
			tree = render(<UserActivationStatus />);
		});

		const { getByTestId, container, getByRole } = tree;

		expect(getByTestId('edit-user-activation-status')).toBeInTheDocument();
		expect(getByTestId('activation-switch')).toBeInTheDocument();
		expect(getByRole('checkbox').checked).toBe(true);
		expect(getByTestId('activation-status-label')).toHaveTextContent('Active');

		expect(container).toMatchSnapshot();
	});

	it('should render UserActivationStatus when user status is inactive', async () => {
		let tree;
		await act(async () => {
			tree = render(<UserActivationStatus />);
		});

		const { getByTestId, container, getByRole } = tree;

		expect(getByTestId('edit-user-activation-status')).toBeInTheDocument();
		expect(getByTestId('activation-switch')).toBeInTheDocument();
		expect(getByRole('checkbox').checked).toBe(false);
		expect(getByTestId('activation-status-label')).toHaveTextContent(
			'Inactive'
		);

		expect(container).toMatchSnapshot();
	});

	it('should activate user on clicking switch', async () => {
		const notificationSpy = jest.spyOn(actualNotification, 'default');

		let tree;
		await act(async () => {
			tree = render(<UserActivationStatus />);
		});

		const { getByTestId, container, getByRole } = tree;

		expect(getByTestId('edit-user-activation-status')).toBeInTheDocument();
		expect(getByRole('checkbox').checked).toBe(false);
		expect(getByTestId('activation-status-label')).toHaveTextContent(
			'Inactive'
		);

		await act(async () => {
			await fireEvent.click(getByRole('checkbox'));
		});
		expect(getByRole('checkbox').checked).toBe(true);
		expect(getByTestId('activation-status-label')).toHaveTextContent('Active');

		expect(updateUserActivationStatus).toHaveBeenCalledWith('test123', true);
		expect(mockedRecoilFn).toHaveBeenCalledWith({
			isUserActive: true,
			name: 'Test User',
			passwordUpdatedAt: '2022-04-30 20:35:35',
			userCreatedAt: '2022-04-30 20:35:35',
			userName: 'test123',
			userRole: 'ADMIN',
		});
		expect(notificationSpy).toHaveBeenCalledWith({
			animationIn: 'animate__slideInRight',
			animationOut: 'animate__slideOutRight',
			position: 'bottom-right',
			message: 'User - test123 has been activated',
			title: 'User activated',
			type: 'success',
		});

		expect(container).toMatchSnapshot();
	});

	it('should deactivate user on clicking switch', async () => {
		const notificationSpy = jest.spyOn(actualNotification, 'default');
		updateUserActivationStatus.mockResolvedValue({
			userName: 'test123',
			updatedActivationStatus: false,
		});
		useRecoilState.mockImplementation(() => {
			return [
				{
					userName: 'test123',
					name: 'Test User',
					isUserActive: true,
					userRole: 'ADMIN',
					userCreatedAt: '2022-04-30 20:35:35',
					passwordUpdatedAt: '2022-04-30 20:35:35',
				},
				mockedRecoilFn,
			];
		});

		let tree;
		await act(async () => {
			tree = render(<UserActivationStatus />);
		});

		const { getByTestId, container, getByRole } = tree;

		expect(getByTestId('edit-user-activation-status')).toBeInTheDocument();
		expect(getByRole('checkbox').checked).toBe(true);
		expect(getByTestId('activation-status-label')).toHaveTextContent('Active');

		await act(async () => {
			await fireEvent.click(getByRole('checkbox'));
		});
		expect(getByRole('checkbox').checked).toBe(false);
		expect(getByTestId('activation-status-label')).toHaveTextContent(
			'Inactive'
		);

		expect(updateUserActivationStatus).toHaveBeenCalledWith('test123', false);
		expect(mockedRecoilFn).toHaveBeenCalledWith({
			isUserActive: false,
			name: 'Test User',
			passwordUpdatedAt: '2022-04-30 20:35:35',
			userCreatedAt: '2022-04-30 20:35:35',
			userName: 'test123',
			userRole: 'ADMIN',
		});
		expect(notificationSpy).toHaveBeenCalledWith({
			animationIn: 'animate__slideInRight',
			animationOut: 'animate__slideOutRight',
			message: 'User - test123 has been deactivated',
			title: 'User deactivated',
			position: 'bottom-right',
			type: 'success',
		});

		expect(container).toMatchSnapshot();
	});

	it('should show error notification when user activation status update fails', async () => {
		const notificationSpy = jest.spyOn(actualNotification, 'default');
		updateUserActivationStatus.mockRejectedValue(new Error('update failed'));
		useRecoilState.mockImplementation(() => {
			return [
				{
					userName: 'test123',
					name: 'Test User',
					isUserActive: true,
					userRole: 'ADMIN',
					userCreatedAt: '2022-04-30 20:35:35',
					passwordUpdatedAt: '2022-04-30 20:35:35',
				},
				mockedRecoilFn,
			];
		});

		let tree;
		await act(async () => {
			tree = render(<UserActivationStatus />);
		});

		const { getByTestId, container, getByRole } = tree;

		expect(getByTestId('edit-user-activation-status')).toBeInTheDocument();
		expect(getByRole('checkbox').checked).toBe(true);
		expect(getByTestId('activation-status-label')).toHaveTextContent('Active');

		await act(async () => {
			await fireEvent.click(getByRole('checkbox'));
		});
		expect(getByRole('checkbox').checked).toBe(true);
		expect(getByTestId('activation-status-label')).toHaveTextContent('Active');

		expect(updateUserActivationStatus).toHaveBeenCalledWith('test123', false);
		expect(notificationSpy).toHaveBeenCalledWith({
			animationIn: 'animate__bounceIn',
			animationOut: 'animate__bounceOut',
			title: 'Failed to update activation status',
			message: 'Please try again',
			type: 'danger',
		});

		expect(container).toMatchSnapshot();
	});
});
