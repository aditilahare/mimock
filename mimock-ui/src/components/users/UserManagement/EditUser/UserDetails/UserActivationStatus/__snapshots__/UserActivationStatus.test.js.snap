// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserActivationStatus should activate user on clicking switch 1`] = `
<div>
  <div
    class="flex items-center align-middle my-8 justify-between"
    data-testid="edit-user-activation-status"
  >
    <div
      class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600 mx-20"
    >
      Activation status
    </div>
    <div
      class="w-1/2 mx-20 flex items-center align-middle"
    >
      <span
        class="MuiSwitch-root IosSwitch-root-21"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-27 MuiSwitch-switchBase IosSwitch-switchBase-22 MuiSwitch-colorSecondary PrivateSwitchBase-checked-28 Mui-checked IosSwitch-checked-25"
          data-testid="activation-switch"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-30 MuiSwitch-input"
              type="checkbox"
              value=""
            />
            <span
              class="MuiSwitch-thumb IosSwitch-thumb-23"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track IosSwitch-track-24"
        />
      </span>
      <div
        class="font-sans font-semibold text-xl mx-4 text-slate-800"
        data-testid="activation-status-label"
      >
        Active
      </div>
    </div>
  </div>
</div>
`;

exports[`UserActivationStatus should deactivate user on clicking switch 1`] = `
<div>
  <div
    class="flex items-center align-middle my-8 justify-between"
    data-testid="edit-user-activation-status"
  >
    <div
      class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600 mx-20"
    >
      Activation status
    </div>
    <div
      class="w-1/2 mx-20 flex items-center align-middle"
    >
      <span
        class="MuiSwitch-root IosSwitch-root-31"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-37 MuiSwitch-switchBase IosSwitch-switchBase-32 MuiSwitch-colorSecondary"
          data-testid="activation-switch"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input-40 MuiSwitch-input"
              type="checkbox"
              value=""
            />
            <span
              class="MuiSwitch-thumb IosSwitch-thumb-33"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track IosSwitch-track-34"
        />
      </span>
      <div
        class="font-sans font-semibold text-xl mx-4 text-slate-300"
        data-testid="activation-status-label"
      >
        Inactive
      </div>
    </div>
  </div>
</div>
`;

exports[`UserActivationStatus should render UserActivationStatus when user status is active 1`] = `
<div>
  <div
    class="flex items-center align-middle my-8 justify-between"
    data-testid="edit-user-activation-status"
  >
    <div
      class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600 mx-20"
    >
      Activation status
    </div>
    <div
      class="w-1/2 mx-20 flex items-center align-middle"
    >
      <span
        class="MuiSwitch-root IosSwitch-root-1"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiSwitch-switchBase IosSwitch-switchBase-2 MuiSwitch-colorSecondary PrivateSwitchBase-checked-8 Mui-checked IosSwitch-checked-5"
          data-testid="activation-switch"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input-10 MuiSwitch-input"
              type="checkbox"
              value=""
            />
            <span
              class="MuiSwitch-thumb IosSwitch-thumb-3"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track IosSwitch-track-4"
        />
      </span>
      <div
        class="font-sans font-semibold text-xl mx-4 text-slate-800"
        data-testid="activation-status-label"
      >
        Active
      </div>
    </div>
  </div>
</div>
`;

exports[`UserActivationStatus should render UserActivationStatus when user status is inactive 1`] = `
<div>
  <div
    class="flex items-center align-middle my-8 justify-between"
    data-testid="edit-user-activation-status"
  >
    <div
      class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600 mx-20"
    >
      Activation status
    </div>
    <div
      class="w-1/2 mx-20 flex items-center align-middle"
    >
      <span
        class="MuiSwitch-root IosSwitch-root-11"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-17 MuiSwitch-switchBase IosSwitch-switchBase-12 MuiSwitch-colorSecondary"
          data-testid="activation-switch"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-20 MuiSwitch-input"
              type="checkbox"
              value=""
            />
            <span
              class="MuiSwitch-thumb IosSwitch-thumb-13"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track IosSwitch-track-14"
        />
      </span>
      <div
        class="font-sans font-semibold text-xl mx-4 text-slate-300"
        data-testid="activation-status-label"
      >
        Inactive
      </div>
    </div>
  </div>
</div>
`;

exports[`UserActivationStatus should show error notification when user activation status update fails 1`] = `
<div>
  <div
    class="flex items-center align-middle my-8 justify-between"
    data-testid="edit-user-activation-status"
  >
    <div
      class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600 mx-20"
    >
      Activation status
    </div>
    <div
      class="w-1/2 mx-20 flex items-center align-middle"
    >
      <span
        class="MuiSwitch-root IosSwitch-root-41"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-47 MuiSwitch-switchBase IosSwitch-switchBase-42 MuiSwitch-colorSecondary PrivateSwitchBase-checked-48 Mui-checked IosSwitch-checked-45"
          data-testid="activation-switch"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input-50 MuiSwitch-input"
              type="checkbox"
              value=""
            />
            <span
              class="MuiSwitch-thumb IosSwitch-thumb-43"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track IosSwitch-track-44"
        />
      </span>
      <div
        class="font-sans font-semibold text-xl mx-4 text-slate-800"
        data-testid="activation-status-label"
      >
        Active
      </div>
    </div>
  </div>
</div>
`;
